Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -335,12 +335,22 @@ function Publish-UpgradedServiceFabricApplication
         }
         catch
         {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisterAppTypeOnUpgradeFailure -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            Write-Host -Message $_.Exception.Message
-            if(!$ApplicationTypeAlreadyRegistered)
+            Write-Host (Get-VstsLocString -Key SFSDK_StartUpgradeFailed -ArgumentList $_.Exception.Message)
+
+            try
```

Comment:
Why try-catch is needed here?

Reply:
Because we want to suppress un-register exception. Task failure message should point to upgrade failure. Un-register failure is just a side effect of upgrade failure.